cdap-ui/app/features/dashboard/controllers/dashboard-ctrl.js

```diff
@@ -3,44 +3,45 @@
  */
 
 angular.module(PKG.name+'.feature.dashboard').controller('DashboardCtrl',
-function ($scope, $state, $dropdown, rDashboardsModel, MY_CONFIG, $alert) {
-
+function ($scope, $state, $dropdown, rDashboardsModel, MY_CONFIG, $alert, $timeout) {
 
   $scope.unknownBoard = false;
   $scope.isEnterprise = MY_CONFIG.isEnterprise;
   $scope.dashboards = rDashboardsModel.data || [];
-  $scope.dashboards.activeIndex = parseInt($state.params.tab, 10) || 0;
-
+  var tab = parseInt($state.params.tab, 10);
+  if (isNaN(tab)) {
+    $scope.dashboards.activeIndex = 'system';
+  } else {
+    $scope.dashboards.activeIndex = tab;
+  }
+  var dropdown;
   /**
    * show a dropdown when clicking on the tab of active dashboard
    * @TODO make a directive instead
    */
   $scope.activeTabClick = function (event, index) {
-    if (index !== $scope.dashboards.activeIndex || !$state.includes('dashboard.user')) {
+    if (index == 'system') {
+      $scope.unknownBoard = true;
+      $state.go('dashboard.standard.cdap');
+      $scope.dashboards.activeIndex = 'system';
       return;
     }
-    var toggle = angular.element(event.target);
-    if(!toggle.hasClass('dropdown-toggle')) {
-      toggle = toggle.parent();
-    }
-
-    var scope = $scope.$new(),
-        dd = $dropdown(toggle, {
-          template: 'assets/features/dashboard/templates/partials/tab-dd.html',
-          animation: 'am-flip-x',
-          trigger: 'manual',
-          prefixEvent: 'dashboard-tab-dd',
-          scope: scope
-        });
 
-    dd.$promise.then(function(){
-      dd.show();
-    });
+    if (index !== $scope.dashboards.activeIndex || !$state.includes('dashboard.user')) {
+      $scope.unknownBoard = true;
+      $state.go('dashboard.user', {tab: index});
+      $scope.dashboards.activeIndex = index;
+      return;
+    }
 
-    scope.$on('dashboard-tab-dd.hide', function () {
-      dd.destroy();
+    $scope.dashboards.forEach(function (dashboard, i) {
+      if (index === i) {
+        dashboard.isopen = !dashboard.isopen;
+      }
```

Comment:
wouldn't this code block equal to just doing $scope.dashboards[index] = !$scope.dashboards[index] ??